services/community-site/src/utils/validators.ts

```diff
@@ -0,0 +1,34 @@
+import { Validator, ValidatorWithStats, YieldedValidator } from '../interfaces/Validator';
+
+const calulcateValidatorYield = (validators: Validator[]): ValidatorWithStats[] | Validator[] => {
+  const hasStats = (validators[0] as ValidatorWithStats)?.pbftsProduced !== 0;
+  if (!hasStats) {
+    return validators;
+  }
```

Comment:
You are assuming that the first validator in the list will not have 0 blocks produced. You can use `every` or `some` here

Reply:
Right, I wanted this to be a check of whether the attribute exists on the object, so an `undefined` check should've come here.